src/Azure.Functions.Cli/Actions/HostActions/StartHostAction.cs

```diff
@@ -480,31 +515,102 @@ public override async Task RunAsync()
             await runTask;
         }
 
+        private async Task<bool> ShouldStartChildProcessFromHostRuntime(bool isInProc, bool isVerbose)
+        {
+            if (string.Equals(HostRuntime, "default", StringComparison.OrdinalIgnoreCase))
+            {
+                if (isInProc)
+                {
+                    throw new CliException($"The runtime host value passed in, default, is not a valid host version for your project. For the default host runtime, the worker runtime must be set to {WorkerRuntime.dotnetIsolated}.");
+                }
+                PrintVerboseForHostSelection(isVerbose, "out-of-process");
+                return false;
+            }
+            else if (string.Equals(HostRuntime, DotnetConstants.InProc8HostRuntime, StringComparison.OrdinalIgnoreCase))
+            {
+                if (!await IsInProcNet8Enabled())
+                {
+                    throw new CliException($"The runtime host value passed in, {DotnetConstants.InProc8HostRuntime}, is not a valid host version for your project. For the {DotnetConstants.InProc8HostRuntime} runtime, the {Constants.FunctionsInProcNet8Enabled} variable must be set while running a .NET 8 in-proc app.");
+                }
+                PrintVerboseForHostSelection(isVerbose, DotnetConstants.InProc8HostRuntime);
+                return true;
+            }
+            else if (string.Equals(HostRuntime, DotnetConstants.InProc6HostRuntime, StringComparison.OrdinalIgnoreCase))
+            {
+                if (await IsInProcNet8Enabled())
+                {
+                    throw new CliException($"The runtime host value passed in, {DotnetConstants.InProc6HostRuntime}, is not a valid host version for your project. For the {DotnetConstants.InProc6HostRuntime} runtime, the {Constants.FunctionsInProcNet8Enabled} variable must not be set while running a .NET 6 in-proc app.");
+                }
+                PrintVerboseForHostSelection(isVerbose, DotnetConstants.InProc6HostRuntime);
+                return true;
+            }
+            // Throw an exception if HostRuntime is set to none of the expected values
+            throw new CliException($"Invalid host runtime '{HostRuntime}'. Valid values are 'default', '{DotnetConstants.InProc8HostRuntime}', '{DotnetConstants.InProc6HostRuntime}'.");
+        }
+
+        private void PrintVerboseForHostSelection(bool isVerbose, string hostRuntime)
+        {
+            if (isVerbose)
+            {
+                ColoredConsole.WriteLine(VerboseColor($"Selected {hostRuntime} host"));
+            }
+        }
+
+        private bool ShouldLaunchChildProcessAfterInferringHostRuntimeAsync(bool shouldLaunchNet8, bool isInProc, bool isVerbose)
+        {
+            // Check if the app is in-proc
+            if (isInProc)
+            {
+                // Start .NET 8 child process if InProc8 is enabled
+                if (shouldLaunchNet8)
```

Comment:
Please simplify to avoid code duplication.

Reply:
added a code block that we only run this logic if we are running an inproc app